build.gradle.kts

```diff
@@ -28,6 +28,25 @@ subprojects {
             include("**/kotlin/**")
         }
     }
+
+    tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
+        sourceCompatibility = JavaVersion.VERSION_1_8.toString()
+        targetCompatibility = JavaVersion.VERSION_1_8.toString()
+
+        kotlinOptions {
+            jvmTarget = JavaVersion.VERSION_1_8.toString()
+            freeCompilerArgs = listOf("-Xopt-in=kotlin.RequiresOptIn")
+            languageVersion = "1.3"
+        }
+    }
+
+    plugins.withId("com.android.application") {
+        val ext = extensions.getByName("android") as com.android.build.gradle.AppExtension
+        ext.compileOptions {
+            sourceCompatibility = JavaVersion.VERSION_1_8
+            targetCompatibility = JavaVersion.VERSION_1_8
+        }
+    }
```

Comment:
let's add a comment that this is only needed if AGP <= 4.1.x since AGP 4.2 is the default value https://developer.android.com/studio/releases/gradle-plugin#java-8-default